### PR TITLE
4.x: metrics and openapi endpoints should be authorized by default

### LINKED ISSUE #7439: 4.x: metrics and openapi endpoints shoudl be authorized by default
Currently we expose `/metrics` without any checks. It would be better to make this endpoint requiring authorization, so users do not accidentaly expose it to the world.
In 4.x we have the new `SecureHandler` that can be used without dependency on Security. See `LogService.routing(HttpRules)` for an example of use.

### PR DESCRIPTION
**Pull request by tvallin**
### Description

fix #7439

### Documentation

no impact

### CODE REVIEW

File: webserver/service-common/src/test/java/io/helidon/webserver/servicecommon/HelidonSecureFeatureTest.java [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+import io.helidon.webserver.http.HttpService;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+
+public class HelidonSecureFeatureTest {
```
Comment (tomas-langer) [inline webserver/service-common/src/test/java/io/helidon/webserver/servicecommon/HelidonSecureFeatureTest.java:35]: Please do not declare test classes as public

File: webserver/tests/observe/metrics/src/test/java/io/helidon/webserver/tests/observe/metrics/ObserveMetricsSecurityTest.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+import io.helidon.webserver.testing.junit5.ServerTest;
+import io.helidon.webserver.testing.junit5.SetUpServer;
+
+import org.junit.jupiter.api.Test;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+@ServerTest
+public class ObserveMetricsSecurityTest {
```
Comment (tomas-langer) [inline webserver/tests/observe/metrics/src/test/java/io/helidon/webserver/tests/observe/metrics/ObserveMetricsSecurityTest.java:46]: just package local please

File: webserver/tests/observe/metrics/src/test/java/io/helidon/webserver/tests/observe/metrics/ObserveMetricsSecurityTest.java [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    private static final Map<String, MyUser> USERS = new HashMap<>();
+
+    private final Http1Client client;
+
+    private ObserveMetricsSecurityTest(URI uri) {
```
Comment (tomas-langer) [inline webserver/tests/observe/metrics/src/test/java/io/helidon/webserver/tests/observe/metrics/ObserveMetricsSecurityTest.java:51]: And please keep the constructor package local as well. Private should be reserved for things that are really internal

File: webserver/tests/observe/metrics/src/test/java/io/helidon/webserver/tests/observe/metrics/ObserveMetricsSecurityTest.java [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+        WebClientSecurity securityService = WebClientSecurity.create(security);
+
+        client = Http1Client.builder()
+                .baseUri(uri)
+                .addService(securityService)
+                .build();
+    }
+
+    @SetUpServer
+    public static void setup(WebServerConfig.Builder server) {
```
Comment (tomas-langer) [inline webserver/tests/observe/metrics/src/test/java/io/helidon/webserver/tests/observe/metrics/ObserveMetricsSecurityTest.java:67]: remove public

File: webserver/tests/observe/metrics/src/test/java/io/helidon/webserver/tests/observe/metrics/ObserveMetricsSecurityTest.java [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+        server.routing(routing -> routing
+                .addFeature(ObserveFeature.create())
+                .addFeature(ContextFeature.create())
+                .addFeature(buildWebSecurity().securityDefaults(SecurityFeature.authenticate()))
+                .get("/observe/metrics", SecurityFeature.rolesAllowed("user")));
+    }
+
+    private static SecurityFeature buildWebSecurity() {
```
Comment (tomas-langer) [inline webserver/tests/observe/metrics/src/test/java/io/helidon/webserver/tests/observe/metrics/ObserveMetricsSecurityTest.java:75]: Please use code formatting from the project, private methods should never be before non-private methods

**Review by tomas-langer — DISMISSED** [review]

**Review by romain-grecourt — DISMISSED** [review]

**Review by romain-grecourt — APPROVED** [review]

**Review by romain-grecourt — APPROVED** [review]

**Pull request merged**